cmd/gossamer/config.go

```diff
@@ -127,11 +138,21 @@ func setDotAccountConfig(ctx *cli.Context, cfg *dot.AccountConfig) {
 
 // setDotCoreConfig sets dot.CoreConfig using flag values from the cli context
 func setDotCoreConfig(ctx *cli.Context, cfg *dot.CoreConfig) {
-	// check --authority flag and update node configuration
-	if authority := ctx.GlobalBool(AuthorityFlag.Name); authority {
-		cfg.Authority = true
-	} else if ctx.IsSet(AuthorityFlag.Name) && !authority {
-		cfg.Authority = false
+	// check --roles flag and update node configuration
+	if roles := ctx.GlobalString(RolesFlag.Name); roles != "" {
+		// convert string to byte
+		b, err := strconv.Atoi(roles)
+		if err != nil {
+			log.Error("[cmd] Failed to convert Roles to byte", "error", err)
+		} else if byte(b) == 4 {
+			// if roles byte is 4, act as an authority (see Table D.2)
+			log.Debug("[cmd] Authority enabled", "roles", 4)
+			cfg.Authority = true
+		} else {
+			// if roles byte is not 4, do not act as an authority (see Table D.2)
```

Comment:
we should have some sort of check for invalid input eg. 3, 5

Reply:
will add error log :+1: